lib/src/utils/size_estimators.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:arweave/src/models/models.dart';
+
+import '../utils.dart';
+import 'bundle_tag_parser.dart';
+
+int estimateDataItemSize({
+  required int fileDataSize,
+  required List<Tag> tags,
+  required List<int> nonce,
+}) {
+  const targetLength = 1;
+  final anchorLength = nonce.isEmpty ? 1 : 1 + 32;
+
+  final serializedTags = serializeTags(
+      tags: tags
+          .map(
+            (t) => Tag(
+              encodeStringToBase64(t.name),
+              encodeStringToBase64(t.value),
```

Comment:
Do we encode the `name` and `value` of the tags ONLY for private entities, right? Public entities are supposed to hold their `name` and `value` in plain text

Reply:
Encoding into Base64 changes the input size

Reply:
https://docs.arweave.org/developers/server/http-api#transaction-format
See tags

Reply:
This is a little confusing to me. Isn't [that](https://docs.arweave.org/developers/server/http-api#transaction-format) saying that the value of `name` is in the domain of `baseURL` and the value of `value` is in the domain of base64URL, instead of asking you to encode the inputs to base64? 

I'm saying this because the tag values we get from GQL are in plain text and in base64 (there's no character outside the base64 domain even though the strings weren't encoded):
```json
            "tags": [
              {
                "name": "Content-Type",
                "value": "application/json"
              },
              {
                "name": "ArFS",
                "value": "0.11"
              },
              {
                "name": "Entity-Type",
                "value": "file"
              },
              {
                "name": "Drive-Id",
                "value": "95b9ac9b-c9e6-4931-b1c5-4867c31b992b"
              },
              {
                "name": "Parent-Folder-Id",
                "value": "ecfef116-43db-44dd-a0dc-855252cf9375"
              },
              {
                "name": "File-Id",
                "value": "9f5ed753-d304-423d-bc7a-6df6400ffabb"
              },
              {
                "name": "App-Name",
                "value": "ArDrive-Web"
              },
              {
                "name": "App-Version",
                "value": "1.3.1"
              },
              {
                "name": "Unix-Time",
                "value": "1640709314"
              }
            ]
```

**Edit**
OR MAYBE... the REST API requires the tags to be encoded before submitting (which I don't know because the ArweaveJS lib would do it on its hand), so we are wrongly estimating the sizes (should we estimate the encoded or non-encoded data?)

Reply:
1. The Arweave lib is the one that makes the encoding under the hood
2. We don't actually care about the size of the tags (but the data) when talking about fees (that was initially my concern)

So I'd say this is just fine